src/pycirclizely/utils/hover_formatter.py

```diff
@@ -0,0 +1,212 @@
+from typing import Dict, List, Optional, Union
+
+import numpy as np
+from plotly.basedatatypes import BaseTraceType
+
+from pycirclizely import config
+from pycirclizely.parser.table import StackedBarTable
+from pycirclizely.patches import PolarSVGPatchBuilder
+from pycirclizely.types import HoverText, NumericSequence
+
+from .helper import deep_dict_update, precise_position
+from .plot import build_scatter_trace
+
+
+class HoverFormatter:
+    hover_texts: List[str]
+    hover_x: List[float]
+    hover_y: List[float]
+    hover_colors: List[str]
```

Comment:
unless other languages, in python this syntax declares *static fields" instead of instance fields. You have 2 options:
1) convert to dataclass. Then you can use this syntax
2) remove and move field declaration to assignment in constructor